slatedb/src/manifest/store.rs

```diff
@@ -76,14 +80,15 @@ impl FenceableManifest {
     ) -> Result<Self, SlateDBError> {
         let clock = system_clock.clone();
         // Initialize generic fenceable record using writer epoch
-        let fr = FenceableRecord::init(
+        let fr = FenceableTransactionalObject::init(
```

Comment:
yeah this makes me a little anxious (re my other comment about making it part of the ops). if you forget to wrap the inner in this you can get unfenced access to the manifest, which seems bad. I feel like its either fenced or not and that should be defined by the protocol rather than the object itself (you shouldn't ever be able to access an object both ways). In fact it'd be nice if the protocol somehow enforced this (objects written with a fencable protocol can't be written by a non-fenced one).

Reply:
I think this discussion goes beyond the scope of this PR. Fundamentally the slatedb protocols allow writing an object with and without fencing. The manifest takes the following kinds of writes:
- writes from writer clients (which all fence each other but not the compactor)
- writes from compactor (which fence each other but not the writer)
- writes from other clients - currently just to add checkpoints